src/greenpak/i2c.py

```diff
@@ -86,3 +86,42 @@ def read(
             data = self.__i2c.read(byte_count)
         self.__i2c.stop()
         return data if ack else None
+
+
+class GreenPackSMBusAdapter(GreenPakI2cInterface):
+    """An adpater to the Linux 'native' SMBus interface"""
+    import smbus2
+
+    def __init__(self, i2cbusdev="/dev/i2c-0"):
+        self.bus = self.smbus2.SMBus(i2cbusdev)
+
+    @override
+    def write(self, addr: int, data: bytearray, silent: bool = False) -> bool:
+        try:
+            if 0 == len(data):
+                #assume caller probing for GP devs on bus
+                self.bus.write_quick(addr)  #Can throw
+            else:
+                #assume data is sent/written
+                self.bus.write_i2c_block_data(addr, data[0], data[1:])
+        except Exception as e:
+            if not silent:
+                str = "SMBus: while writing to add 0x%02x: caught exception: %s" % (addr, e)
+                print(str)
+            return False
+        return True
+
+    @override
+    def read(
+        self, addr: int, byte_count: int, silent: bool = False
+    ) -> bytearray | None:
+        offset = 0
+        data = []
+        while byte_count > 0:
```

Comment:
The semantic here seems to be different than the other two drivers here. Can you make it to make a zero bytes read if byte_count is zero?  (similar to what you did in the write method?).

Reply:
Ok, I will also check other two drivers code to understand why attempts to read 0 bytes  are used.
With 0-writes i saw that is used for probing/finding greepaks.

Reply:
I was not sure what is better/safer to scan the I2C bus, empty writes or empty reads, so left both options open.

Reply:
Extra comments added in code specifically to address it.  (No, empty read attempts, on smbus/i2c are not safe, thus the underlying layer provides empty writes only. However, see latest code too).